contracts/mainnet/connectors/socket/main.sol

```diff
@@ -0,0 +1,89 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.7.6;
+pragma experimental ABIEncoderV2;
+
+/**
+ * @title Socket.
+ * @dev Multi-chain Bridge Aggregator.
+ */
+
+import { Basic } from "../../common/basic.sol";
+import { TokenInterface } from "../../common/interfaces.sol";
+import "./events.sol";
+import "./interface.sol";
+
+abstract contract SocketConnectorResolver {
+
+    /**
+     * @dev Gets Allowance target from registry.
+     * @param _route route number
+     */
+    function getAllowanceTarget(uint _route) internal view returns (address _allowanceTarget) {
+        ISocketRegistry registryContr = ISocketRegistry(0xc30141B657f4216252dc59Af2e7CdB9D8792e1B0);
+        ISocketRegistry.RouteData memory data = registryContr.routes(_route);
+        return data.route;
+    }
+}
+
+abstract contract SocketConnector is SocketConnectorResolver, Basic {
+
+    address constant registry = 0xc30141B657f4216252dc59Af2e7CdB9D8792e1B0;
+
+    struct BridgeParams {
+        bytes txData;
+        address token;
+        uint256 amount;
+	}
+
+    /**
+	 * @dev Bridge Token.
+	 * @notice Bridge Token on Socket.
+	 * @param _token token address on source chain
+     * @param _txData tx data for calling
+     * @param _route route number
+     * @param _amount amount to bridge
+	 * @param _getId ID to retrieve amount from last spell.
+	 */
+    function bridge (
+        address _token,
+        bytes memory _txData,
+        uint256 _route,
+        uint256 _amount,
+        uint256 _getId
+    )
+        external payable returns (string memory _eventName, bytes memory _eventParam) 
+    {
+        _amount = getUint(_getId, _amount);
```

Comment:
@shriyatyagii I think we can't use getId because we already have the `_amount` in the `_txData`

Reply:
yes, got your point. Removed